### DISCUSSION TITLE
Fixing "Module not found: Can't resolve 'fs'" Error

### DISCUSSION THREAD

**Discussion by gtsui** [wevm/wagmi#2379] [General]
I'm trying to run wagmi@1.0.5 in a nextjs@13.4.2 application and running into the below error. Any ideas what I'm doing wrong here and how do I fix this?

Error: Module not found: Can't resolve 'fs'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/node-gyp-build/index.js
./node_modules/bufferutil/index.js
./node_modules/ws/lib/buffer-util.js
./node_modules/ws/lib/websocket.js
./node_modules/ws/index.js
./node_modules/@walletconnect/legacy-client/dist/esm/socket.js
./node_modules/@walletconnect/legacy-client/dist/esm/core.js
./node_modules/@walletconnect/legacy-client/dist/esm/index.js
./node_modules/@walletconnect/legacy-provider/dist/esm/signer.js
./node_modules/@walletconnect/legacy-provider/dist/esm/index.js
./node_modules/@wagmi/core/node_modules/@wagmi/connectors/dist/walletConnectLegacy.js
./node_modules/@wagmi/core/dist/connectors/walletConnectLegacy.js
./node_modules/wagmi/dist/connectors/walletConnectLegacy.js

**Comment by tmm** [comment]
Hi! You might want to provide more info, like a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) so folks can easily help out.

**Reply by Jam516** [reply]
Hi I got this same error. Here are some details @tmm 

**Describe the bug**

I added wagmi config to my NextJS project following the example given here https://github.com/family/connectkit/blob/main/examples/nextjs/pages/_app.tsx

When I run my app I get this error
./node_modules/node-gyp-build/node-gyp-build.js:1:0
Module not found: Can't resolve 'fs'

Import trace for requested module:
./node_modules/node-gyp-build/index.js
./node_modules/bufferutil/index.js
./node_modules/ws/lib/buffer-util.js
./node_modules/ws/lib/websocket.js
./node_modules/ws/index.js
./node_modules/@walletconnect/legacy-client/dist/esm/socket.js
./node_modules/@walletconnect/legacy-client/dist/esm/core.js
./node_modules/@walletconnect/legacy-client/dist/esm/index.js
./node_modules/@walletconnect/legacy-provider/dist/esm/signer.js
./node_modules/@walletconnect/legacy-provider/dist/esm/index.js
./node_modules/@wagmi/connectors/dist/walletConnectLegacy.js
./node_modules/@wagmi/core/dist/connectors/walletConnectLegacy.js
./node_modules/wagmi/dist/connectors/walletConnectLegacy.js
./node_modules/connectkit/build/index.es.js
./pages/_app.tsx

The project works when I remove the config from app.tsx (seen in screenshot)

**To reproduce**

Here is the Replit 
project https://replit.com/@0xKofi/BazaarFrontendFork?v=1

connectkit@1.4.0
viem@0.3.43
wagmi@1.0.9

**Screenshots**
![Screenshot 2023-06-01 175153](https://github.com/wagmi-dev/wagmi/assets/15823285/b02db302-545b-46ab-b5a2-bb4e8396f499)
